### PR TITLE
Feat/instrument unit testflags

### PR DESCRIPTION
**Pull request by frrist**
### Description
This PR adds unit and functional test flags to the appropriate tests

### CODE REVIEW

File: proofs/sectorbuilder/testing/interface_test.go [hunk 1]
```diff
@@ -28,6 +29,9 @@ const MaxTimeToSealASector = time.Second * 360
 const MaxTimeToGenerateSectorPoSt = time.Second * 360
 
 func TestSectorBuilder(t *testing.T) {
+	tf.FunctionalTest(t)
+
+	// TODO DO NOT MERGE THERE BE DRAGONS!! WHAT THIS BE?
```
Comment (travisperson) [inline proofs/sectorbuilder/testing/interface_test.go]: _Comment in reference to the following code (for discussion so you don't have to scroll so far :-)_
```go
if os.Getenv("FILECOIN_RUN_SECTOR_BUILDER_TESTS") != "true" {
	t.SkipNow()
}
```

I remember the reason for this was to skip this test in CI and have it run instead on a schedule due to it's very expensive execution time (issue #1976). I think the question is do we want to keep this pattern or extend the the flag pattern in some way.

Comment (frrist) [reply proofs/sectorbuilder/testing/interface_test.go]: Thanks for pointing this out. To remain consistent with other tests I'd like to replace the env var with a flag. Maybe something like `SectorTest` or `ProofsTest`, then [modify](https://github.com/filecoin-project/go-filecoin/blob/2bfb648484544f2a3b3f3f2e8afbe3d26a6fb2a1/.circleci/config.yml#L233) CI runs to work with it as it does now.

Comment (anorth) [reply proofs/sectorbuilder/testing/interface_test.go]: I thought that existing functional tests already run only occasionally, but may be misremembering. I don't think we need a configuration that doesn't run in dev workflows but does run always in CI. Is that what `-functional` is now? I think we want doesn't run in dev workflows or regular CI, but does run occasionally. We could repurpose the functional flag for that. Let's get rid of the env vars.

Don't block this PR on figuring that out, let's discuss elsewhere.

**Comment by ognots** [comment]
Awesome work!
I only have one request - can the final commit message describe how this changes the invocation of tests. 
Also maybe some documentation in the README?

**Comment by frrist** [comment]
@ognots I have updated the commit message to better reflect the changes. As the code stands now tests still run the same, in a follow on PR I will update the docs and changes the flags so they are not all on by default.

**Pull request merged**